PWGHF/HFC/TableProducer/correlatorFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,324 @@
+
+#include <CCDB/BasicCCDBManager.h>
+#include <CommonConstants/MathConstants.h>
+#include <CommonConstants/PhysicsConstants.h>
+#include <Framework/ASoA.h>
+#include <Framework/ASoAHelpers.h>
+#include <Framework/AnalysisDataModel.h>
+#include <Framework/AnalysisHelpers.h>
+#include <Framework/AnalysisTask.h>
+#include <Framework/BinningPolicy.h>
+#include <Framework/Configurable.h>
+#include <Framework/GroupedCombinations.h>
+#include <Framework/HistogramRegistry.h>
+#include <Framework/HistogramSpec.h>
+#include <Framework/InitContext.h>
+#include <Framework/Logger.h>
+#include <Framework/OutputObjHeader.h>
+#include <Framework/runDataProcessing.h>
+
+#include <string>
+#include <vector>
+
+using namespace o2;
+using namespace o2::hf_centrality;
+using namespace o2::hf_evsel;
+
+enum DecayChannel {
+  DplusToPiKPi = 0,
+  DsToKKPi,
+  DsToPiKK
+};
+
+/// Code to select collisions with at least one Ds meson
+struct HfCorrelatorFlowCharmHadrons {
+  Produces<aod::HfcRedFlowColls> rowCollisions;
+  Produces<aod::HfcRedCharmHads> rowCharmCandidates;
+  Produces<aod::HfcRedCharmMls> rowCharmCandidatesMl;
+  Produces<aod::HfcRedTrkAssoc> rowAssocTrackReduced;
+  Produces<aod::HfcRedTrkSels> rowAssocTrackSelInfo;
+
+  Configurable<int> centEstimator{"centEstimator", 2, "Centrality estimation (FT0A: 1, FT0C: 2, FT0M: 3, FV0A: 4)"};
+  Configurable<int> selectionFlag{"selectionFlag", 1, "Selection Flag for hadron (e.g. 1 for skimming, 3 for topo. and kine., 7 for PID)"};
+  Configurable<bool> forceCharmInCollision{"forceCharmInCollision", false, "Flag to force charm in collision"};
+  Configurable<std::string> ccdbUrl{"ccdbUrl", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
+  Configurable<std::vector<int>> classMl{"classMl", {0, 2}, "Indexes of BDT scores to be stored. Two indexes max."};
+  Configurable<float> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<float> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+  Configurable<float> ptCandMax{"ptCandMax", 24., "max. cand. pT"};
+  Configurable<float> etaTrackMax{"etaTrackMax", 1., "max. track eta"};
+  Configurable<float> ptTrackMin{"ptTrackMin", 0.15, "min. track pT"};
+  Configurable<float> ptTrackMax{"ptTrackMax", 5., "max. track pT"};
+  Configurable<float> dcaXYTrackMax{"dcaXYTrackMax", 1., "max. track DCA XY"};
+  Configurable<float> dcaZTrackMax{"dcaZTrackMax", 1., "max. track DCA Z"};
+
+  HfHelper hfHelper;
+  HfEventSelection hfEvSel; // event selection and monitoring
+  o2::framework::Service<o2::ccdb::BasicCCDBManager> ccdb;
+  SliceCache cache;
+
+  double massCharm{0.};
+
+  using CollsWithCentMult = soa::Join<aod::Collisions, aod::EvSels, aod::FT0Mults, aod::FV0Mults, aod::CentFT0Ms, aod::CentFT0As, aod::CentFT0Cs, aod::CentFV0As>;
+  using CandDsDataWMl = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi, aod::HfMlDsToKKPi>>;
+  using CandDsData = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDsToKKPi>>;
```

Comment:
Personally I would remove the rectangualr-selection-based process functions and the relative filtered table. We will anyway use BDTs for these studies (from pp or OO when the MC will be out)